subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -2374,24 +2366,36 @@ void lwm2m_udp_receive(struct net_context *ctx, struct net_pkt *pkt,
 	}
 }
 
-static void udp_receive(struct net_context *ctx, struct net_pkt *pkt,
+static void udp_receive(struct net_app_ctx *app_ctx, struct net_pkt *pkt,
 			int status, void *user_data)
 {
-	lwm2m_udp_receive(ctx, pkt, pendings, NUM_PENDINGS,
-			  replies, NUM_REPLIES, false, handle_request);
+	struct lwm2m_ctx *client_ctx = CONTAINER_OF(app_ctx,
+						    struct lwm2m_ctx,
+						    net_app_ctx);
+
+	lwm2m_udp_receive(client_ctx, pkt, false, handle_request);
 }
 
 static void retransmit_request(struct k_work *work)
 {
+	struct lwm2m_ctx *client_ctx;
 	struct zoap_pending *pending;
 	int r;
 
-	pending = zoap_pending_next_to_expire(pendings, NUM_PENDINGS);
+	client_ctx = CONTAINER_OF(work, struct lwm2m_ctx, retransmit_work);
+	if (!client_ctx) {
```

Comment:
Hi @mike-scott ,

I left two comments yesterday.
One probably folded because I left the comment on the outdated part of the code.

I check the macro "CONTAINER_OF", it's a simple offset deduction based on the given memory address. I'm suggesting to remove the checking here for it's unlikely the address will be NULL.
```
if (!client_ctx) {
       ...
}
```

Reply:
Thanks.  Fixing.